### PR TITLE
Migrate to Gulp4 && Fix `yo yeoman-boilerplate-es6` not found

### PR DESCRIPTION
**Pull request by zckrs**
Note: I unable NSP task because is really long to respond

### CODE REVIEW

File: test/index.js [hunk 1]
```diff
@@ -1,8 +1,28 @@
-import assert from 'assert';
-import sampleNode from '../generators/app';
+import { join as pathsJoin } from 'path';
+import { test as helpers } from 'yeoman-generator';
+import assert from 'yeoman-assert';
 
-describe('generator-yeoman-boilerplate-es6', function () {
-  it('should have unit test!', function () {
+describe('yeoman-boilerplate-es6:app', () => {
+  before(function (done) {
```
Comment (zckrs) [inline test/index.js]: @Swiip arrow function doesn't work here `before(done => {`
Do you know why ?

**Comment by Swiip** [comment]
:+1:

File: test/index.js (hunk 1 above)
Comment (Swiip) [inline test/index.js]: Yep! Mocha (Jasmine the same) assign your context before launching the function. If you use an arrow function, you break the managed context of the function.

Comment (zckrs) [inline test/index.js]: Thank your for infos.

**Pull request merged**